### ISSUE TITLE
What's a right way to include normal directions or color components in addition to positions (x,y,z)?

### ISSUE THREAD

**Issue by neon5d** [rusty1s/pytorch_geometric#180]
## ❓ Questions & Help

I am looking for the right way to add normal directions or color components in addition to positions (x,y,z). I can't find any example.

Is it ok just to increase the dimension of the position like the below pseudo code?
pos = ([torch.tensor(data['vertex'][axis]) for axis in ['x', 'y', 'z']]) 
nn =   ([torch.tensor(data['vertex'][axis]) for axis in ['nx', 'ny', 'nz']]) 
cc = ([torch.tensor(data['vertex'][axis]) for axis in ['r', z', 'b']])
pos = torch.cat((pos, nn, cc), 1)
data = Data(pos=pos)

**Comment by rusty1s** [comment]
Hi, we generally refer to `x` as our input features, but you can enhance data objects by any attribute you want, e.g., `data.norm` or `data.color`, or save everything in `data.x` by default. I do not recommend adding any other information to `data.pos` as their position due to semantic reasons. Hope this makes things clearer.
